app/models/beta/search/goods_nomenclature_query.rb

```diff
@@ -89,19 +89,20 @@ def goods_nomenclature_item_id_term_query
       def multi_match_query
         candidate_query = { size:, query: { bool: {} } }
 
-        candidate_query[:query][:bool][:filter] = filter_part if filters.any?
+        candidate_query[:query][:bool][:filter] = filter_part
         candidate_query[:query][:bool][:must] = must_part if must_part.any?
         candidate_query[:query][:bool][:should] = should_part if should_part.any?
 
         candidate_query
       end
 
       def filter_part
-        {
-          bool: {
-            must: Api::Beta::GoodsNomenclatureFilterGeneratorService.new(@filters).call,
-          },
-        }
+        part = { bool: {} }
+
+        part[:bool][:must] = static_and_dynamic_filters
+        part[:bool][:must] << declarable_filter
+
+        part
```

Comment:
Since this method set only `must` part of the filter, should this be called `must_part`? 🤔 
But actually there is another method called `must_part`, which apparently does not set `must` ... strange 🤔

Reply:
Spoke offline and added some context for this